firestore/src/jni/compare.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2021 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Returns the result of `Object::Equals` for the given `lhs` and `rhs` in the
+ * current Firestore `jni::Env`.
+ */
+template <typename T>
+bool CompareJni(const T& lhs, const T& rhs) {
+  Env env = FirestoreInternal::GetEnv();
```

Comment:
Nit: add the appropriate includes.

Reply:
Drive-by comment: You should just create a new `Env` object and use it here. The `Env` object is tied to the thread that created it and by using the `Env` returned from `FirestoreInternal::GetEnv()` it makes this comparison cause undefined behavior if ever used on a different thread (e.g. the user callback thread).

Reply:
Alternately, you could require the caller to specify an `Env&` reference as an argument, with is semi-conventional within our code base.

Reply:
Done.